shell/platform/embedder/embedder.h

```diff
@@ -0,0 +1,102 @@
+// Copyright 2017 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+#ifndef FLUTTER_EXPORT
+#define FLUTTER_EXPORT
+#endif  // FLUTTER_EXPORT
+
+#define FLUTTER_ENGINE_VERSION 1
+
+typedef enum {
+  kSuccess = 0,
+  kInvalidLibraryVersion,
+  kInvalidArguments,
+} FlutterResult;
+
+typedef enum {
+  kOpenGL,
+} FlutterRendererType;
+
+typedef void* FlutterEngine;
+
+typedef bool (*BoolCallback)(void* /* user data */);
+typedef uint32_t (*UIntCallback)(void* /* user data */);
+
+typedef struct {
+  BoolCallback make_current;
+  BoolCallback clear_current;
+  BoolCallback present;
+  UIntCallback fbo_callback;
+} FlutterOpenGLRendererConfig;
+
+typedef struct {
+  FlutterRendererType type;
+  union {
+    FlutterOpenGLRendererConfig open_gl;
+  };
+} FlutterRendererConfig;
+
+typedef enum {
+  kWindowMetrics,
+  kPointer,
+} FlutterEventType;
+
+typedef struct {
+  size_t width;
+  size_t height;
+} FlutterWindowMetricsEvent;
+
+typedef enum {
+  kCancel,
+  kUp,
+  kDown,
+  kMove,
+} FlutterPointerPhase;
+
+typedef struct {
+  FlutterPointerPhase phase;
+  size_t timestamp;  // in microseconds
+  double x;
+  double y;
+} FlutterPointerEvent;
+
+typedef struct {
+  FlutterEventType type;
+  union {
+    FlutterWindowMetricsEvent window_metrics;
+    FlutterPointerEvent pointer;
+  };
+} FlutterEvent;
```

Comment:
Rather than making this union, I'd just have different entry points for each type of event.  The receiver is going to need to dispatch them based on type anyway.